packages/web-app-files/tests/unit/components/SideBar/Links/PublicLinks/LinkInfo.spec.js

```diff
@@ -0,0 +1,233 @@
+  linkIndirect: '.oc-files-file-link-via',
+  linkViaLabel: '.files-file-links-link-via-label',
+  linkRole: '.oc-files-file-link-role',
+  linkCopyUrl: '.oc-files-public-link-copy-url'
+}
+const localVue = createLocalVue()
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+/**
+ * There are some errors that should be caught but sneaked by vue.
+ * Vue Warn errors is one of them
+ *
+ * Global console contains these error from where we can assert if they are expected errors
+ */
+console.error = function(message) {
+  if (typeof message === 'string' && message.includes('Vue warn')) {
+    // keep default behaviour
+    // error.apply(console, arguments)
+    // Vue warn contains messages with warning initiator element
+    // Only vue warn message is thrown leaving the initiator element description
+    const errorSplit = message.split('\n')
+    throw errorSplit[0]
+  } else throw message instanceof Error ? message : new Error(message)
+}
+
+describe('LinkInfo', () => {
+  function getShallowWrapper(
+    link = {
+      token: '122235445488',
+      url: 'some-link'
+    }
+  ) {
+    return shallowMount(LinkInfo, {
+      localVue,
+      stubs: { ...stubs, 'oc-tag': true },
+      propsData: {
+        link: link
+      },
+      directives: {
+        'oc-tooltip': jest.fn()
+      }
+    })
+  }
+
+  function getMountedWrapper(
+    link = {
+      token: '122235445488',
+      url: 'some-link'
+    }
+  ) {
+    localVue.use(DesignSystem)
+    return mount(LinkInfo, {
+      localVue,
+      propsData: {
+        link: link
+      },
+      stubs,
+      directives: {
+        'oc-tooltip': jest.fn()
+      }
+    })
+  }
+
+  describe('link name', () => {
+    it('should show link name as token if link does not have name', () => {
+      const wrapper = getShallowWrapper()
+      expect(wrapper.vm.linkName).toEqual('122235445488')
+      expect(wrapper.find(selectors.linkName).text()).toEqual('122235445488')
+    })
+
+    it('should show link name if provided link has name attribute', () => {
+      const wrapper = getShallowWrapper({
+        name: 'some-name',
+        url: 'some-link'
+      })
+      expect(wrapper.vm.linkName).toEqual('some-name')
+      expect(wrapper.find(selectors.linkName).text()).toEqual('some-name')
+    })
+
+    it('should show link name if provided link has both name and token attribute', () => {
+      const wrapper = getShallowWrapper({
+        name: 'some-name',
+        token: 'some-token',
+        url: 'some-link'
+      })
+      expect(wrapper.vm.linkName).toEqual('some-name')
+      expect(wrapper.find(selectors.linkName).text()).toEqual('some-name')
+    })
+  })
+  describe('link prop', () => {
+    it('should be required', () => {
+      expect(() => {
+        shallowMount(LinkInfo, {
+          localVue,
+          stubs: { ...stubs, 'oc-tag': true }
+        })
+      }).toThrow('[Vue warn]: Missing required prop: "link"')
+    })
```

Comment:
Should we test Vue errors? I am not sure.
cc @individual-it

Reply:
this basically checks if `required: true` is set, so I think its good the test it. If the behavior would change and someone would make it not required it might introduce a bug